internal/kafka/internal/services/clusters_test.go

```diff
@@ -2336,7 +2472,480 @@ func Test_ClusterService_GetExternalID(t *testing.T) {
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "fake-strimzi-version",
+			},
+			want:    false,
+			wantErr: false,
+		},
+	}
+	g := NewWithT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := clusterService{
+				connectionFactory: tt.fields.connectionFactory,
+				providerFactory:   tt.fields.providerFactory,
+			}
+			got, err := c.CheckStrimziVersionReady(tt.args.cluster, tt.args.strimziVersion)
+			g.Expect(got).To(Equal(tt.want))
+			g.Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_clusterService_IsStrimziKafkaVersionAvailableInCluster(t *testing.T) {
+	type fields struct {
+		connectionFactory *db.ConnectionFactory
+		providerFactory   clusters.ProviderFactory
+	}
+	type args struct {
+		cluster        *api.Cluster
+		strimziVersion string
+		kafkaVersion   string
+		ibpVersion     string
+	}
+
+	strimziOperatorVersion := "strimzi-cluster-operator.from-cluster"
+	availableStrimziVersions, err := json.Marshal([]api.StrimziVersion{
+		{
+			Version: strimziOperatorVersion,
+			Ready:   true,
+			KafkaVersions: []api.KafkaVersion{
+				{
+					Version: "2.7.0",
+				},
+				{
+					Version: "2.8.0",
+				},
+			},
+			KafkaIBPVersions: []api.KafkaIBPVersion{
+				{
+					Version: "2.7",
+				},
+				{
+					Version: "2.8",
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		t.Fatal("failed to convert available strimzi versions to json")
+	}
+
+	false_availableStrimziVersions, err := json.Marshal([]api.StrimziVersion{
+		{
+			Version: strimziOperatorVersion,
+			Ready:   false,
+		},
+	})
+
+	if err != nil {
+		t.Fatal("failed to convert available strimzi versions to json")
+	}
+
+	mockCluster := &api.Cluster{
+		Meta: api.Meta{
+			ID:        testID,
+			CreatedAt: time.Now(),
+		},
+		Region:                   testKafkaRequestRegion,
+		ClusterID:                testClusterID,
+		CloudProvider:            testKafkaRequestProvider,
+		Status:                   api.ClusterReady,
+		AvailableStrimziVersions: availableStrimziVersions,
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "should return true if Strimzi kafka versions are available in cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-cluster-operator.from-cluster",
+				kafkaVersion:   "2.7.0",
+				ibpVersion:     "2.7",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "Returns false if the provided kafka and kafka ibp versions are not available in the cluster",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args: args{
+				cluster:        mockCluster,
+				strimziVersion: "strimzi-version",
```

Comment:
strimzi version here should be valid

```suggestion
				strimziVersion: strimziOperatorVersion,
```